test/src/test_register.rs

```diff
@@ -232,3 +234,57 @@ fn test_advanced_methods() -> bool {
 
     ok
 }
+
+#[derive(NativeClass)]
+#[inherit(Reference)]
+#[register_with(VarargsToTuple::register)]
+struct VarargsToTuple {}
+
+#[methods]
+impl VarargsToTuple {
+    fn new(_owner: TRef<Reference>) -> Self {
+        VarargsToTuple {}
+    }
+
+    fn register(builder: &ClassBuilder<VarargsToTuple>) {
+        builder.method("calc", CalcMethod).done();
+    }
+}
+
+struct CalcMethod;
+
+impl Method<VarargsToTuple> for CalcMethod {
+    fn call(
+        &self,
+        _this: TInstance<'_, VarargsToTuple>,
+        args: gdnative::export::Varargs<'_>,
+    ) -> Variant {
+        let (a, b, c): (i64, i64, i64) =
+            std::convert::TryInto::try_into(args).expect("Must be able to convert");
+        let ret = a * b - c;
+        ret.to_variant()
+    }
+}
+
+fn test_varargs_to_tuple() -> bool {
+    println!(" -- test_varargs_to_tuple");
+
+    let ok = std::panic::catch_unwind(|| {
+        let thing = Instance::<VarargsToTuple, _>::new();
+        let base = thing.base();
+
+        assert_eq!(Some(7), unsafe {
+            base.call(
+                "calc",
+                &[3_i64.to_variant(), 4_i64.to_variant(), 5_i64.to_variant()],
+            )
+            .to()
+        });
```

Comment:
Very complex expression, could you extract some parts into variables?

Also, we try to use the argument order convention `assert_eq!(actual, expected)` -- even if it's not yet consistent everywhere. So `Some(7)` would be at the end. If the assert itself is one line, this is not less readable.

Reply:
Like this?
```suggestion
        let args = [3_i64.to_variant(), 4_i64.to_variant(), 5_i64.to_variant()];
        assert_eq!(unsafe { base.call("calc", &args).to() }, Some(7));
```

Reply:
Yes, that looks great! 🙂